zeppelin-plugins/notebookrepo/gcs/src/test/java/org/apache/zeppelin/notebook/repo/GCSNotebookRepoTest.java

```diff
@@ -187,6 +187,28 @@ public void testRemove() throws Exception {
     assertThat(storage.get(makeBlobId(runningNote.getId(), runningNote.getPath()))).isNull();
   }
 
+  @Test
+  public void testRemoveFolder_nonexistent() throws Exception {
+    try {
+      notebookRepo.remove("id", "/name", AUTH_INFO);
+      fail();
+    } catch (IOException e) {}
```

Comment:
try catch block is not necessary.

Reply:
If I removed the try catch block, then the test would fail.

The purpose of this test is to make sure that the function would throw an exception if it cannot remove the folder, resulting in the `fail()` part of the test to never be reached.

Reply:
Perhaps the following article will help you. This shows you how your test expects an exception.
https://www.baeldung.com/junit-assert-exception

I would prefer 
`@Test(expected = IOException.class)`

Reply:
I see

That can do

I was following the previous tests written in the class

Changing it.

Reply:
@Reamer 
Updated
Please check again
Thanks!